docs/_guides/working-with-events.md

```diff
@@ -0,0 +1,109 @@
+with configuration matching the one provided in the request. Pod is wrapped in the deployment, so that the pod will be
+rescheduled/restarted if needed. See [Registering custom event driver](custom-event-drivers.md) for more details.
+
+## How to use event driver in your environment
+
+### Adding event driver
+
+You can use Dispatch CLI or API to add an event driver (UI support coming soon). In following example, we're adding 
+new vCenter event driver. 
+
+```
+dispatch create event-driver vcenter --name vcenter1.corp.local
+```
+
+The above will create new event driver of type `vcenter` and --name `vcenter1.corp.local`. Creating an event-driver
+takes a while - monitor its status by running:
+
+```
+dispatch get event-driver vcenter1.corp.local
+```
+
+When status changes to `RUNNING`, event driver is ready to use.
+
+Event driver by itself doesn't do much. it produces events which are ingested by Dispatch, but Dispatch does not know what
+to do with them yet. To tell Dispatch to react to events, you need to create a subscription.
+
+## Working with subscriptions
+### Adding subscription
+
+Subscription tells Dispatch to match certain events to certain function, so that when events matching defined conditions arrive
+to the system, these functions will be executed. Events that do not match any subscriptions are automatically dropped.
+
+To add a new subscription for `vm.being.created` event in vcenter event driver, run:
+
+```
+dispatch create subscription --event-type vm.being.created myFunction
+```
+
+The above command will print output similar to the following:
+
+```
+           NAME          | SOURCE TYPE |    EVENT TYPE    | FUNCTION NAME | STATUS |         CREATED DATE
+---------------------------------------------------------------------------------------------------------
+  complete-cicada-410962 | *           | vm.being.created | myFunctio     | READY  | Fri Dec 31 17:18:54 PST -0001
+```
+
+The above subscription will cause dispatch to execute function `myFunction` for every event of type `vm.being.created`. 
+Note the asterisk in  `SOURCE TYPE` column. This is default value and say "match all source Types".
+What is source type? If you create event driver of `vcenter` type, `vcenter` is your source type. 
+
+You can also specify a name for your subscription using `--name` parameter. if you don't, a random, human-readable name will be created.  
+
+### Event driver event types
+
+To find out the list of event types produced by built-in event drivers, see [Built-in Event Drivers](built-in-event-drivers.md).
+
+
+## Emitting events via CLI or API
+
+Event driver is a powerful concept, but there are cases when you want to emit an event by your self or through some external application.
+Dispatch exposes dedicated API to emit events.
+
+To emit an event using CLI, you can use `dispatch emit` subcommand.
+
+In a following example, We emit an event of type `my.event`:
```

Comment:
In _the_ following example.

This is one article change that should be made to avoid confusion.